File: KCD/HMRealDestroyShipCommand.m
//
//  HMRealDestroyShipCommand.m
//  KCD
//
//  Created by Hori,Masaki on 2014/07/20.
//  Copyright (c) 2014年 Hori,Masaki. All rights reserved.
//

#import "HMRealDestroyShipCommand.h"

#import "HMServerDataStore.h"

@implementation HMRealDestroyShipCommand
- (void)execute
{
	HMServerDataStore *store = [HMServerDataStore oneTimeEditor];
	
	NSString *destroyedShipId = [self.arguments objectForKey:@"api_ship_id"];
	
	NSError *error = nil;
	NSArray<NSManagedObject *> *ships = [store objectsWithEntityName:@"Ship"
															   error:&error
													 predicateFormat:@"id = %@", @([destroyedShipId integerValue])];
	if(ships.count == 0) {
		return;
	}
	[store deleteObject:ships[0]];
}
@end

File: KCD/HMRealDestroyShipCommand.swift
//
//  HMRealDestroyShipCommand.swift
//  KCD
//
//  Created by Hori,Masaki on 2015/01/18.
//  Copyright (c) 2015年 Hori,Masaki. All rights reserved.
//

import Cocoa

class HMRealDestroyShipCommand: HMJSONCommand {
	override func execute() {
		if let destroyedShipID = arguments["api_ship_id"] as? String {
			let store = HMServerDataStore.oneTimeEditor()
			var error: NSError? = nil
			let predicate = NSPredicate(format: "id = \(destroyedShipID)")
			let ships = store.objectsWithEntityName("Ship",
				sortDescriptors: nil,
				predicate: predicate,
				error: &error)
			if(ships.count == 0) { return }
			if let ship = ships[0] as? HMKCShipObject {
				store.managedObjectContext!.deleteObject(ship)
			}
		}
	}
}
